src/WebJobs.Script.WebHost/Controllers/HostController.cs

```diff
@@ -287,6 +290,33 @@ public IActionResult GetAdminToken()
             return Ok(requestHeaderToken);
         }
 
+        [HttpPost]
+        [Route("admin/warmup")]
+        [Authorize(Policy = PolicyNames.AdminAuthLevelOrInternal)]
+        [RequiresRunningHost]
+        public async Task<IActionResult> Warmup([FromServices] IScriptJobHost scriptHost, [FromServices] IScriptHostManager scriptHostManager, [FromServices] IHostIdProvider hostIdProvider)
+        {
+            // The function must be called warmup and should be of type warmup trigger
+            FunctionDescriptor function = scriptHost.GetFunctionOrNull(WarmUpConstants.FunctionName);
+            if (function != null && function.Metadata.ContainsInputBinding(WarmUpConstants.WarmupTrigger))
+            {
+                var hostStatus = new HostStatus
```

Comment:
I think we need a strongly typed contract here, e.g. a WarmupContext that we new up and invoke the function with. Need to decide what data to pass initially as part of this context. Not sure State is useful...

Reply:
Chatted offline. I agree with a WarmupContext. But, having a hard time figuring out the right info. If anyone has suggestions, that would help.

Also wondering if something like "InstanceCount" would have made sense. This info would have to come from the caller. So we can't add that now.

Reply:
For now, we will have an empty `WarmupContext` object. fyi, @fabiocav